### ISSUE TITLE
Samsung S10 & Android 10

### ISSUE THREAD

**Issue by Dragancokan** [NightscoutFoundation/xDrip#1122]
Hi

Yesterday I updated my Samsung S10 to Android 10

Unfortunately, Xdrip does not work anymore
I use Dexcom G6
I have the latest version of 5.12.2019 on the run and also the app settings for system settings is turned on unfortunately it does not work. Does anyone have a tip or idea?

Thanks and regards

Dragan

**Comment by lbarmen** [comment]
Too I suffer all day, I do not receive data, already tried all settings.

Samsung s10
android 10
dexcom g5

**Comment by playingbball20** [comment]
I have tried everything, and still hoping there is something.  I am using the Dexcom app.  You have to get the custom app....
 
Build Your Own Dexcom App
https://docs.google.com/forms/d/e/1FAIpQLScD76G0Y-BlL4tZljaFkjlwuqhT83QlFM5v6ZEfO7gCU98iJQ/viewform?usp=sf_link

**Comment by lbarmen** [comment]
no, nightscout is important to me, I'd rather try going back to android 9.

**Comment by checkatillo** [comment]
Nightscout still can be used through Dexcom Share

**Comment by iqhu** [comment]
I have the same problem with my S10 & G6 since the Android 10 update. I have tried through all xdrip settings combinations and guides and fixes, without success (and lost my last 6m of xdrip data, luckily it is uploaded to Nightscout). Reading the logs, it seems to have to do with BLE. xdrip was able to read the sensor only once every time I forgot it and restarted the collector, but not again thereafter. Funnily enough, the Dexcom app has no problems.

I have now been successfully using the following workaround, similar to what the person above me says: The official Dexcom app reads the G6 with Share enabled. On my Nightscout Heroku app, I have set up Dexcom Share as the source for BG readings. Then, in the xdrip app, I have set up Dexcom Share as a BG source, while also syncing my treatments to Nightscout. So in the end, I have the same functionality as before (e.g. using xdrip to manage my diabetes and having everything uploaded to Nightscout), with the drawback that xdrip sometimes misses the readings from Share for a few minutes (but its usually at max. 1 reading behind) and it is dependent on internet connection. But even without internet, the Dexcom app is running in the background and can be used.

Let us hope that either a new Samsung or xdrip update fixes this soon. If anyone has success, please share your strategy.

**Comment by TimothyGold** [comment]
I am having the same problems using anything newer than xDrip-plus-20190413-05bd730.apk. I am still able to use this version but automatic restarts don't work anymore. If I want more than 10 days I have to go through a full manual restart and warmup cycle. I keep trying the new versions but everything new fails to get raw data from the sensor.

**Comment by troubar** [comment]
I got the same problem, after updating my S10e to android 10
Thanks for pointing out the old apk that works. I will try it out, and hope it works consistantly.

**Comment by mtih1992** [comment]
I had this same issue. The following steps worked for me:

Turn off bluetooth on mobile
force close xDrip
Restart phone
Turn on bluetooth.


xDrip version: d7689af-2019.11.02

**Comment by BKrueckeberg** [comment]
Hi! I am facing the same issue but the steps that @mtih1992 posted did not work for me. I installed the official Dexcom app instead. It works but sucks. Is anyone working on fixing this?

**Comment by troubar** [comment]
After playing around with the different apk-versions and the different connection settings, nothing worked for me. I also went to the dexcom app. 

After a fresh install of xdrip, I get a connection to my G6 after few readings (always less than 5) it stops working again. (sadly I forgot to take any screenshots from the logs)

**Comment by playingbball20** [comment]
I know I'm a few days behind, but have you used the newest version?  It fixed it and have had no errors since.

**Comment by BKrueckeberg** [comment]
I was able to get it working today. I installed the most recent build from January 9 and started a new G6 sensor. After the warmup everything was fine. Very happy to have xdrip on my phone and watch again... Love it!

**Comment by dhansen7** [comment]
After I upgraded my Samsung phone to Android 10, xDrip stopped working. This turned my life upside down.

I installed Nightly build 16th Jan 2020 "xDrip-plus-20200116-8c4987f.apk" on Samsung S10+ using Android 10.

xDrip has been working for several hours without any issues.

THANK YOU!   xDrip has become a very important part of my life.

**Comment by jesaca7** [comment]
Hi dhansen7 what settings have you use, got the same problem.
Dexcom g5.
Updated yesterday to android 10 on my S10+. Then I updated xdrip to the version found here: https://github.com/NightscoutFoundation/xDrip/releases/tag/2020.01.16 and tried several settings but no luck at the moment. When I add the G5 It get the first readings I calibrate it but then It lost the connection and couldnt gather readings any more.

I am downloading android 9 and I probably do a roll back if the workarounds didnt work.

Thanks!

**Comment by dhansen7** [comment]
Hello Jesaca7,

I'm sorry that you too are having problems with xDrip with the Dexcom G5
system on your Samsung S10 using Android 10.

Unfortunately, my Samsung S10 has stopped working again so I don't have any
advice that would help you.

Here's what I can tell you.

I installed xDrip on my Samsung S10 when it was using Android 9. Everything
was working great for a month until I upgraded to Android 10.

After the upgrade, xDrip only worked for a day or so. I called Dexcom
support thinking that it was a problem with the G5 transmitter.

Dexcom replaced my transmitter and sensor. xDrip and my Samsung S10 worked
for 4 days before stopping again.

I researched the posts on NightScout's website and tried the suggestions I
found. I tried different xDrip versions going back to April 2019. I tried
flashing my phone back to Android Pie 9. Flashing did not work because I'm
using a Verizon locked phone. I had to do a factory reset which wiped my
phone of all my programs, data, photos, music, basically everything.

Even though I had backed everything up, it still took me a week to get
everything back the way it was before the reset.

I reinstalled xDrip. No success even while trying several different
versions.

After reading a post about someone having success with Android 10 using the
latest xDrip version, I upgraded back to Android 10. I installed the latest
version of xDrip.

Success!

But for only 4 days.

I called Dexcom Support  who determined that my sensor was bad because both
my Dexcom Receiver and phone were getting the same faulty inconsistent
readings.

I installed a new sensor last night but have not started using xDrip again,
just my Dexcom Receiver.

To summarize:

- All of these issues started with the Android 10 upgrade.

- Trying different versions did not resolve the problem.

- Going back to Android 9 with a factory reset did not resolve the problem.

- Trying different versions again did not resolve the problem.

- Upgrading back to Android 10 with the latest xDrip version (1/16) worked
for 4 days.

Since my latest issue may have involved a faulty sensor, I will need to
test further.

I wish I had some encouraging information for you. I will be updating my
NightScout post today. If I find a solution, I'll email you.

Regards,
David

**Comment by jesaca7** [comment]
Hi David,
I have completed the roll back to Android 9 and using now the latest xdrip is working on my S10+ I am still configuring the phone and everything but at least the sensor is sending the data and the Bluetooth link is up.
I followed the steps on this video https://youtu.be/hVFpzGdvLNM to complete the downgrade bad lucky enough xdrip started working afterwards.
I am using the same sensor and transmitter than I have at the beggining.

Thanks 
Jesus

**Comment by dhansen7** [comment]
That's the same video I used to downgrade but my phone would not accept the
Android 9 file. It didn't matter though because a factory reset essentially
provided the same downgrade.

Good luck. I hope you are successful.

David

**Comment by dhansen7** [comment]
I found some information that might help in Issue #1185 posted yesterday.

Link:
https://github.com/NightscoutFoundation/xDrip/issues/1185

**Comment by dhansen7** [comment]
I found some information that might help in Issue #1185 posted yesterday.

Link:
https://github.com/NightscoutFoundation/xDrip/issues/1185

**Comment by billymild** [comment]
Has anyone fixed the issue with success?

**Comment by mtih1992** [comment]
Try rolling back to a version dated 31/12/2019. I couldn't connect on the
newest versionns but trial and error, rolling back to this one made it
connect again.

Might have been a fluke though!

**Comment by billymild** [comment]
I am running the 1-24-2020 nightly build. I have unchecked some settings in bluetooth settings. It seems to be more  consisten capture, but I won't know how good until I use it for a while.

**Comment by dhansen7** [comment]
I'm using a Samsung S10+ with Android 10 with a G5 sensor/transmitter.

I installed xDrip version 2020.01.24 from the nightly versions. I used the
suggested Bluetooth settings mentioned in Issue #1185. xDrip has now been
working perfectly for over 2 days.

In the past, I've used the Dexcom Receiver and my phone simultaneously.
This time, I never turned on the receiver.

Hope this works for you.

**Comment by Dragancokan** [comment]
I only get values ​​every 15-20 minutes. No matter what Xdrip version or settings

**Comment by dhansen7** [comment]
Are you removing the cache and data before uninstalling the old version of
xDrip before installing the latest version?

I had problems until I removed the old data.

**Comment by billymild** [comment]
@dhansen7 I did not remove cache or data last time. I actually restored from backup this last time. But I had solid readings for  a LONG time yesterday, then at night after disconnecting from my watch things went south.

**Comment by mtih1992** [comment]
Found another workaround;

"Forget Bluetooth device in xdrip.
Go to Settings - Apps - See all Apps - (3 dots on top right corner) - Show system apps - Bluetooth - Delete Data

Then you can try to connect with your miao miao again."

Taken from this thread:

https://github.com/NightscoutFoundation/xDrip/issues/1001

Seems to have successfully got my miaomiao to show as connected in xdrip

**Comment by mtih1992** [comment]
This is also on the latest nightly build (26th Jan 2020) - followed above workaround and it connected successfully, recieved data and is working. 

Will report back tomorrow about how long-term this fix is.

**Comment by jamorham** [comment]
I've received some logs from a S10+ user which show that samsung are blocking the xDrip wake up timing. This is the sort of thing we've seen from them before with non-standard power saving features.  For anyone having problems I would look again if you can find any other hidden power saving whitelist for apps etc.

**Comment by MasterPlexus** [comment]
Due to that, as this is also known from OnePlus: Samsung 10+ could test if it helps to lock xdrip in the recent apps list (open recent apps, scroll to xdrip picture, then there should be a open lock, or a possible menu, where the app could be locked). I have seen that this lock was checked by the additional poversaving routines of some phone firmwares. Maybe this helps...

**Comment by mtih1992** [comment]
> Found another workaround;
> 
> "Forget Bluetooth device in xdrip.
> Go to Settings - Apps - See all Apps - (3 dots on top right corner) - Show system apps - Bluetooth - Delete Data
> 
> Then you can try to connect with your miao miao again."
> 
> Taken from this thread:
> 
> #1001
> 
> Seems to have successfully got my miaomiao to show as connected in xdrip

UPDATE: This has worked for me. I've also changed power settings in my S10 to stop battery optimisation. I've turned off the "Adaptive Battery" settings in android settings, and also turned off "Put unused apps to sleep".

**Comment by Dragancokan** [comment]
I reset my s10 with Android 10 to factory settings today.  xdrip and AndroidAPS installed in current versions.  since then both apps work great.  I'm so happy

**Comment by billymild** [comment]
@Dragancokan have you noticed longer battery life?

**Comment by Dragancokan** [comment]
no I did not. But my smartphone held out all day

**Comment by rodrigowebjump** [comment]
I disabled the Trust Auto-Connect in Less Common Settings-> BlueTooth Settings and it connected successfully.

**Comment by JZhovner** [comment]
Turn off "Use the OB1 Collector". it helped me

**Comment by iqhu** [comment]
After using a workaround since the issue came up after the S10 update back then, by using the Dexcom app as collector and then using Share to get the values into xdrip, I am now back to normal. Since a few days neither xdrip nor Nightscout could get the Dexcom Share values, so I tried using my S10 with xdrip as collector again. And it works! I have the newest firmware installed (as of 20 March 2020) and the latest xdrip nightly. No issues for 2 days now.
